### DISCUSSION TITLE
Alerting: Allow matching of alert status in Notification policies

### DISCUSSION THREAD

**Discussion by bentasker** [grafana/grafana#64551] [Alerting]
Currently, Notification Policies can only match on labels.

So, for example, I can easily say: Send alerts with `host=foo` to contact `bar`.

However, this will result in all statuses going to `bar` which may not be desirable - `bar` may want to receive `Firing` and `Resolved` but not `DataSourceError`.

As a real-world example, I have the following setup:

* Root notification policy: Send via email
* Notification Policy: match `severity=critical` and send to PagerDuty (to generate a page)

However, if my datasource is unavailable (for example, down for a full filesystem backup) the alert will go to status `DataSourceError` and generate a page.

This might be quite unwelcome at 2am. Mute timings can't be used for this, because you **do** want to receive a page if the thing being monitored is genuinely on fire.

It is possible to override this behaviour on the alert:

![override_state](https://user-images.githubusercontent.com/2900301/224118902-80f0a28c-5928-474c-8124-1854aec59830.png)

But that's not necessarily desirable:

* It means editing every alert *and* remembering to set this on any future alerts
* It'll change the alert reference/key for anything downstream that relies on a consistent label set
* You may still want an email notification to note that the interruption happened

It might be possible to hack something similar together by using a string template to set a label value (using `{{ .Status }}`) but it'd be nice to have a way that's less involved and doesn't require customising every current and future alert.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). We are migrating this discussion to an issue and closing the discussion. The issue is #82845. Feel free to continue the discussion around this there. Thank you!
